bin/ch/WScriptJsrt.cpp

```diff
@@ -1371,6 +1372,48 @@ JsValueRef __stdcall WScriptJsrt::SleepCallback(JsValueRef callee, bool isConstr
     return returnValue;
 }
 
+JsValueRef __stdcall WScriptJsrt::GetProxyPropertiesCallback(JsValueRef callee, bool isConstructCall, JsValueRef *arguments, unsigned short argumentCount, void *callbackState)
+{
+    HRESULT hr = E_FAIL;
+    JsValueRef returnValue = JS_INVALID_REFERENCE;
+    JsErrorCode errorCode = JsNoError;
+
+    IfJsrtErrorSetGo(ChakraRTInterface::JsGetUndefinedValue(&returnValue));
+
+    if (argumentCount > 1)
+    {
+        bool isProxy = false;
+        JsValueRef target;
+        JsValueRef handler;
+        ChakraRTInterface::JsGetProxyProperties(arguments[1], &isProxy, &target, &handler);
+
+        if (isProxy)
+        {
+            JsPropertyIdRef targetProperty;
+            JsPropertyIdRef handlerProperty;
+            
+            IfJsrtErrorSetGo(CreatePropertyIdFromString("target", &targetProperty));
+            IfJsrtErrorSetGo(CreatePropertyIdFromString("handler", &handlerProperty));
+            IfJsrtErrorSetGo(ChakraRTInterface::JsCreateObject(&returnValue));
+
+            if (target == nullptr)
+            {
+                JsValueRef undefined = JS_INVALID_REFERENCE;
+                IfJsrtErrorSetGo(ChakraRTInterface::JsGetUndefinedValue(&undefined));
+                IfJsrtErrorSetGo(ChakraRTInterface::JsSetProperty(returnValue, targetProperty, undefined, true));
+                IfJsrtErrorSetGo(ChakraRTInterface::JsSetProperty(returnValue, handlerProperty, undefined, true));
```

Comment:
Not sure if there is a benefit to doing this here -- the value of return.target will be undefined regardless of whether or not you set it. I personally would say that this method should return `{ target, handler}` or `undefined` rather than `{target, handler}` or `{}`

Reply:
Note changed - I think there is potential utility to this - also it enables me to explicitly test the handling of a revoked proxy.

Currently this API does the following:
1. if given a non-proxy -> returns undefined
2. if given a revoked proxy -> returns { target : undefined, handler : undefined} -> you can explicitly detect a revoked proxy this way as the return value will have two properties not 0
3. if given a proxy that has not been revoked returns {target : proxyTarget, handler : proxyHandler}

Reply:
Sure, but then if nothing else, you still dont need to assign undefined to ret.target and ret.handler, they'll be undefined regardless.

Reply:
@jackhorton They will, but then `Object.keys(ret).length` would then be 0 instead of 2 which seems to be what @rhuanjl was getting at.

Reply:
I guess it could have {} as the return value for a revoked proxy - just seemed easier to explicitly test it this way.

Reply:
The object returned from this helper can have a `revoked` property set to true if handler and target are null. We could use that to detect a revoked proxy. Would be easier to test that. Pretty sure real proxy can never have undefined or null handler and target unless it's revoked, though. I'm not really invested in either solution.